src/main/java/org/dependencytrack/tasks/OSVDownloadTask.java

```diff
@@ -0,0 +1,174 @@
+
+            try {
+                for (Ecosystem ecosystem : Ecosystem.values()) {
+                    URL url = new URL("https://osv-vulnerabilities.storage.googleapis.com/"+ ecosystem.getValue() +"/all.zip");
+                    ZipInputStream zipIn = new ZipInputStream(url.openStream());
+                    unzipFolder(zipIn);
+                    zipIn.closeEntry();
+                }
+            } catch (IOException exception) {
+                exception.printStackTrace();
+            }
+        }
+    }
+
+    private void unzipFolder(ZipInputStream zipIn) throws IOException {
+
+        BufferedReader reader;
+        GoogleOSVAdvisoryParser parser = new GoogleOSVAdvisoryParser();
+        try {
+            ZipEntry zipEntry = zipIn.getNextEntry();
+            while (zipEntry != null) {
+
+                reader = new BufferedReader(new InputStreamReader(zipIn));
+                String line = null;
+                StringBuilder out = new StringBuilder();
+                while ((line = reader.readLine()) != null) {
+                    out.append(line);
+                }
+                JSONObject json = new JSONObject(out.toString());
+                System.out.println(json);
+                final OSVAdvisory osvAdvisory = parser.parse(json);
+                updateDatasource(osvAdvisory);
+                zipEntry = zipIn.getNextEntry();
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void updateDatasource(final OSVAdvisory advisory) {
+        LOGGER.info("Updating datasource with Google OSV advisories");
+        try (QueryManager qm = new QueryManager()) {
+
+            LOGGER.debug("Synchronizing Google OSV advisory: " + advisory.getId());
+            final Vulnerability synchronizedVulnerability = qm.synchronizeVulnerability(mapAdvisoryToVulnerability(qm, advisory), false);
+            final List<VulnerableSoftware> vsList = new ArrayList<>();
+            for (OSVVulnerability osvVulnerability: advisory.getVulnerabilities()) {
+                VulnerableSoftware vs = mapVulnerabilityToVulnerableSoftware(qm, osvVulnerability);
+                if (vs != null) {
+                    vsList.add(vs);
+                }
+            }
+            LOGGER.debug("Updating vulnerable software for OSV advisory: " + advisory.getId());
+            qm.persist(vsList);
+            synchronizedVulnerability.setVulnerableSoftware(vsList);
+            qm.persist(synchronizedVulnerability);
+        }
+        Event.dispatch(new IndexEvent(IndexEvent.Action.COMMIT, Vulnerability.class));
+    }
+
+    private Vulnerability mapAdvisoryToVulnerability(final QueryManager qm, final OSVAdvisory advisory) {
+
+        final Vulnerability vuln = new Vulnerability();
+        vuln.setSource(Vulnerability.Source.GOOGLE);
+        vuln.setVulnId(String.valueOf(advisory.getId()));
+        vuln.setTitle(advisory.getSummary());
```

Comment:
OSV provides only summary and details fields (https://ossf.github.io/osv-schema/#summary-details-fields). It says "_It is recommended that this field be kept short, on the order of no more than 120 characters_" but its no hard rule so it can be longer. So yes, we can do either hard cut or customise the title.

Reply:
I'm making it hard cut to length 253 and then appending 2 dots, if length of summary is exceeding max length 255.